formats/properties/commonMain/src/kotlinx/serialization/properties/Properties.kt

```diff
@@ -89,6 +89,22 @@ public sealed class Properties(
             return structure(descriptor).also { copyTagsTo(it) }
         }
 
+        final override fun <T> decodeSerializableValue(deserializer: DeserializationStrategy<T>): T {
+            val type = map["type"]?.toString()
+            var actualSerializer: DeserializationStrategy<out Any>? = null
+
+            if (type != null && deserializer is AbstractPolymorphicSerializer<*>) {
+                actualSerializer = deserializer.findPolymorphicSerializerOrNull(this, type)
+            }
+
+            if (actualSerializer == null) {
+                throw RuntimeException()
```

Comment:
You can use `findPolymorphicSerializer` without 'orNull'. It will throw a `SerializationException` with a proper message for you.